### PR TITLE
feat: added comment report table and migration

### PR DESCRIPTION
**Pull request by capJavert**
Add comment_report table and migration.

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]
Looks alright to me!

File: src/entity/CommentReport.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  commentId: string;
+
+  @PrimaryColumn({ length: 36 })
+  @Index('IDX_comment_report_user_id')
+  userId: string;
+
+  @Column({ default: () => 'now()' })
+  createdAt: Date;
+
+  @Column({ length: 12 })
```
Comment (sshanzel) [inline src/entity/CommentReport.ts:17]: We have a report reason that pertains to misinformation and has thought of using `MISINFORMATION` as its key. The limit is 12, but the said key is 14 character. We can maybe extend the length to 16? WDYT?
```suggestion
  @Column({ length: 16 })
```

File: src/migration/1687956058691-CommentReport.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class CommentReport1687956058691 implements MigrationInterface {
+    name = 'CommentReport1687956058691'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "comment_report" ("commentId" text NOT NULL, "userId" character varying(36) NOT NULL, "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "reason" character varying(12) NOT NULL, "note" text, CONSTRAINT "PK_cef0f7f862ea6ffd5a99dd640eb" PRIMARY KEY ("commentId", "userId"))`);
```
Comment (sshanzel) [inline src/migration/1687956058691-CommentReport.ts:7]: In relation to the initial comment about extending the length.
```suggestion
        await queryRunner.query(`CREATE TABLE "comment_report" ("commentId" text NOT NULL, "userId" character varying(36) NOT NULL, "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "reason" character varying(16) NOT NULL, "note" text, CONSTRAINT "PK_cef0f7f862ea6ffd5a99dd640eb" PRIMARY KEY ("commentId", "userId"))`);
```

File: src/entity/CommentReport.ts (hunk 1 above)
Comment (rebelchris) [reply src/entity/CommentReport.ts:17]: Oh good catch, could even be safe and make it 36 doens't add much overhead I personally think.

File: src/migration/1687956058691-CommentReport.ts (hunk 2 above)
Comment (capJavert) [reply src/migration/1687956058691-CommentReport.ts]: Added `36` as Chris suggested.

**Pull request merged**